include/picongpu/plugins/openPMD/openPMDWriter.hpp

```diff
@@ -998,9 +997,51 @@ Please pick either of the following:
                     fieldsOffsetDims[0] = globalOffsetFile;
                 }
 
+                auto const componentSize = field_no_guard.productOfComponents();
+
                 /* write the actual field data */
                 for(uint32_t d = 0; d < nComponents; d++)
                 {
+                    ::openPMD::MeshRecordComponent mrc
```

Comment:
Why is this code moved into the loop? Was it a bug and before the meta data was added to the wrong object?

Reply:
The diff looks a bit misleading on this one. The code is still at the same level, but now it's in the beginning of the iteration. The braces at 1071/2 close a contained loop.